src/Grid/CalendarView.vala

```diff
@@ -196,17 +196,26 @@ public class Maya.View.CalendarView : Gtk.Grid {
     /* Sets the calendar widgets to the date range of the model */
     void sync_with_model () {
         var model = Model.CalendarModel.get_default ();
-        DateTime previous_first = null;
+
+        // Handle existing grid, if one exists
         if (days_grid != null) {
-            if (days_grid.grid_range != null && (model.data_range.equals (days_grid.grid_range) || days_grid.grid_range.first_dt.compare (model.data_range.first_dt) == 0))
+            assert (days_grid.grid_range != null);
+            if (model.data_range.equals (days_grid.grid_range) || days_grid.grid_range.first_dt.compare (model.data_range.first_dt) == 0)
                 return; // nothing to do
-            if (days_grid.grid_range != null)
-                previous_first = days_grid.grid_range.first_dt;
+
+            // Transition in the proper direction
+            var previous_first = days_grid.grid_range.first_dt;
+            if (previous_first.compare (days_grid.grid_range.first_dt) == -1) {
+                stack.transition_type = Gtk.StackTransitionType.SLIDE_UP;
+            } else {
+                stack.transition_type = Gtk.StackTransitionType.SLIDE_DOWN;
+            }
```

Comment:
This does not work - you are comparing two things that are the same so the comparison yields `0`.  The transition is always `SLIDE_DOWN`.

Reply:
😬️ Yeah, you're right. I forgot that it's being set part way through... I'm just going to undo that commit and leave the minimal change.